forem/forem
# Admin member index: invitations responsive layout

## What type of PR is this? (check all applicable)


## Description

This PR adds an alternative layout for the invitations data, to make it more responsive. Given the invitations table contains less data columns than the main member index, I've made the choice to start showing the full table at the `medium` breakpoint (768px). We might revisit this in a future phase.

## Related Tickets & Documents



Closes https://github.com/forem/forem/issues/16960

(We have an optional final task on the above issue, so I will check with the pod whether we want to commit to completing that before closing it or not)

## QA Instructions, Screenshots, Recordings

- Make sure you have the `member_index_view` feature flag enabled
- Visit `/admin/invitations`
- Set your screen size to a width below 768px. Check the layout against [the designs](https://www.figma.com/file/Ejc7VELuN00n1SZrkUTBcQ/Member-List-View?node-id=709%3A9025) 
- Check the actions in the overflow menu are functional
- Check the search feature is functional
- Increase your screen size until you see the full table, and double-check the look and the actions 

### Screenshots

#### Mobile (XS size)

![mobile layout](https://user-images.githubusercontent.com/20773163/163354074-f4cec79c-fefe-4ba3-80ec-ff6c9c736822.png)

#### Small size (matches mobile, but the "invite member" button is prevented from expanding the full width)

![small layout](https://user-images.githubusercontent.com/20773163/163354177-e317f16d-a601-4d58-af6f-cc448b7e82e3.png)

#### Medium size and above

![full table layout](https://user-images.githubusercontent.com/20773163/163354244-1b7ef694-0fea-466d-bf4e-a47e20c4168f.png)


### UI accessibility concerns?

- Making this responsive improves the accessibility, allowing users to easily zoom in on sections
- I've updated the actions dropdown etc to make sure we have different IDs for each context they're used in

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._


- [X] This change does not need to be communicated, and this is why not: part of a larger issue, behind a feature flag


## [optional] What gif best describes this PR or how it makes you feel?

![tetris](https://media.giphy.com/media/nTYLJY9SXKMqQ/giphy.gif)
